packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm

```diff
@@ -244,26 +244,50 @@ - (void)getRectWithAttributedString:(AttributedString)attributedString
 
 #pragma mark - Private
 
-- (NSTextStorage *)_textStorageForNSAttributesString:(NSAttributedString *)attributedString
+- (NSTextStorage *)_textStorageForNSAttributesString:(NSAttributedString *)inputAttributedString
                                  paragraphAttributes:(ParagraphAttributes)paragraphAttributes
                                                 size:(CGSize)size
 {
-  NSTextContainer *textContainer = [[NSTextContainer alloc] initWithSize:size];
 
-  textContainer.lineFragmentPadding = 0.0; // Note, the default value is 5.
-  textContainer.lineBreakMode = paragraphAttributes.maximumNumberOfLines > 0
-      ? RCTNSLineBreakModeFromEllipsizeMode(paragraphAttributes.ellipsizeMode)
-      : NSLineBreakByClipping;
-  textContainer.maximumNumberOfLines = paragraphAttributes.maximumNumberOfLines;
+  NSMutableAttributedString *attributedString = [ inputAttributedString mutableCopy];
+
+  /*
+    * The block below is responsible for setting the exact height of the view in lines
+    * Unfortunatelly, iOS doesn't export any easy way to do it. So we set maximumNumberOfLines
+    * prop and then add random lines at the front. However, they are only used for layout
+    * so they are not visible on the screen. This method is used for drawing only for Paragraph component
+    * but we set exact height in lines only on TextInput that doesn't use it. 
+    */
+  if (paragraphAttributes.numberOfLines) {
+    paragraphAttributes.maximumNumberOfLines = paragraphAttributes.numberOfLines;
+    NSMutableString *newLines = [NSMutableString stringWithCapacity: paragraphAttributes.numberOfLines];
+    for (NSUInteger i = 0UL; i < paragraphAttributes.numberOfLines; ++i) {
+      // K is added on purpose. New line seems to be not enough for NTtextContainer
+      [newLines appendString:@"K\n"];
```

Comment:
Same comment as above for Android (but this looks different in the paper case of doing this?).

Reply:
it wasn't working with empty lines for some reason. Similarly to Android.